### ISSUE TITLE
client, volume, filer different data stat and volume.balance -force loop

### ISSUE THREAD

**Issue by htsnorman** [seaweedfs/seaweedfs#4695]
**Describe the bug**

I have experienced multiple errors while investigating how storage sizes are reflected in the statistics from the client, server, and seaweed volume sides.

It is evident that the fuse mount calculates the maximum storage size as "volumeSizeLimitMB(master) * max(volume) * seaweed volume node number."

However, when I started downloading files using multiple threads and copying them to the mounted fuse share, I noticed that the client-side returned false values several times. The filer service provided a stat value of 0 bytes or significantly deviated from the actual occupied storage space.

Eventually, I encountered a situation where one of the nodes ran out of memory, causing the volume service to crash due to memory depletion. I restarted it and thought of performing the usual commands:

"lock
volume.balance -force
volume.fix.replication
unlock"

However, even after several hours, it did not complete, and I kept getting the same logs from the "volume.balance -force" command, continuously receiving "replicating volume x from .... to datanode ....." entries. I also attempted to restart the master service, but running it on the other master led to the same loop without ever finishing.

**System Setup**

Services:
- weed volume -dir=/mnt/vol1 -port=8080 -mserver=seaweed-a:9333,seaweed-b:9333,seaweed-c:9333 -dataCenter datacenter-01 -rack rack-01 -fileSizeLimitMB 1000 -max=8
- weed master -ip=seaweed-a -port=9333 -mdir=/opt/seaweedfs -peers=seaweed-a:9333,seaweed-b:9333,seaweed-c:9333 -defaultReplication=001 -volumeSizeLimitMB=1000
- weed filer -port=8888 -port.readonly=8899 -master=seaweed-a:9333,seaweed-b:9333,seaweed-c:9333 -dataCenter=datacenter-01

- Debian 11.7
- version 30GB 3.54 358b3a489475333e503886513080b815ce45a4a5 linux amd64

**Screenshots and Expected behavior**
du -sh show right sizes
ls -la is ok
but df -h show 0 in mounted seaweedfs fuse
there may be a problem because all our monitoring uses the df command
![image](https://github.com/seaweedfs/seaweedfs/assets/9657774/5f123825-d268-4e2f-a2c3-5ed719487c38)

filer too show right data and downloadable all content.
![image](https://github.com/seaweedfs/seaweedfs/assets/9657774/9d5668d9-9f1e-4c0e-a7dc-976dd1baea47)

I don't quite understand the consumption from the volume side, because at the file level it's roughly ~8.6GB, but the 3 volumes only have ~5GB of reserved space.
![image](https://github.com/seaweedfs/seaweedfs/assets/9657774/fa392332-b96e-4dc1-9db1-86dc73864942)
![image](https://github.com/seaweedfs/seaweedfs/assets/9657774/27bb1894-9f4e-4052-9755-86d3038e275c)
![image](https://github.com/seaweedfs/seaweedfs/assets/9657774/bbce4735-0955-4c00-97ed-a41769e6f62b)

the volume storage values are also interesting on the volume gui interface
![image](https://github.com/seaweedfs/seaweedfs/assets/9657774/0e81e1ab-41b1-4a60-a1ff-5bef2092088a)
![image](https://github.com/seaweedfs/seaweedfs/assets/9657774/64522733-d95e-4e3b-a812-c10f9ba10677)
![image](https://github.com/seaweedfs/seaweedfs/assets/9657774/6e4d0104-e304-4509-850c-77005347c7d2)

and the volume.balance -force loop:

```Jul 19 16:51:55 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:51:55 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 19 16:52:10 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
...
...
...
Jul 20 11:01:15 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:15 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:15 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:15 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:15 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:15 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:15 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
Jul 20 11:01:30 sofalvin-seaweedfs-a seaweedfs-master[801]: replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...```


Am I configuring something wrong?

**Comment by htsnorman** [comment]
stuck here:

`I0720 16:40:54.724378 masterclient.go:209 master localhost:9333 redirected to leader seaweed-c:9333
.master: localhost:9333 filers: [192.168.40.45:8888 192.168.40.44:8888 192.168.40.62:8888]
> lock
> volume.balance
Running in simulation mode. Use "-force" option to apply the changes.
hdd 0.79 1.00:0.75	  moving  volume 9 192.168.40.45:8080 => 192.168.40.62:8080
> volume.fix.replication -n
replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
> volume.fix.replication
replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 9 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 11 001 from 192.168.40.44:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...
replicating volume 7 001 from 192.168.40.45:8080 to dataNode 192.168.40.62:8080 ...`

**Comment by htsnorman** [comment]
volume.check.disk:

volume.check.disk -v

Running in simulation mode. Use "-force" option to apply the changes.
load collection  volume 6 index size 7984 from 192.168.40.62:8080 ...
load collection  volume 6 index size 1008 from 192.168.40.44:8080 ...
volume 6 192.168.40.44:8080 has 63 entries, 192.168.40.62:8080 missed 2 and partially deleted 0 entries
volume 6 192.168.40.62:8080 has 285 entries, 192.168.40.44:8080 missed 10 and partially deleted 0 entries
load collection  volume 1 index size 1056 from 192.168.40.62:8080 ...
load collection  volume 1 index size 912 from 192.168.40.44:8080 ...
volume 1 192.168.40.44:8080 has 57 entries, 192.168.40.62:8080 missed 1 and partially deleted 0 entries
sync volume 1 on 192.168.40.62:8080 and 192.168.40.44:8080: doVolumeCheckDisk source:192.168.40.44:8080 target:192.168.40.62:8080 volume 1: rpc error: code = Unknown desc = read needle blob offset 104968680 size 314572830: EOF
>

**Comment by chrislusf** [comment]
there are multiple issues reported here. Need to resolve them one by one.

what is the output of `volume.list`?

maybe you can add some println to see what's going on in this file

https://github.com/seaweedfs/seaweedfs/blob/master/weed/shell/command_volume_balance.go

**Comment by htsnorman** [comment]
volume.list:

volume.list
Topology volumeSizeLimit:1000 MB hdd(volume:19/24 active:19 free:5 remote:0)
  DataCenter datacenter-01 hdd(volume:19/24 active:19 free:5 remote:0)
    Rack rack-01 hdd(volume:19/24 active:19 free:5 remote:0)
      DataNode 192.168.40.44:8080 hdd(volume:6/8 active:6 free:2 remote:0)
        Disk hdd(volume:6/8 active:6 free:2 remote:0)
          volume id:1 size:1228736 file_count:43 replica_placement:1 version:3 modified_at_second:1689846063
          volume id:6 size:573935304 file_count:73 replica_placement:1 version:3 compact_revision:2 modified_at_second:1689864456
          volume id:8 size:2047816 file_count:17 replica_placement:1 version:3 modified_at_second:1689777348
          volume id:10 size:1400560 file_count:12 replica_placement:1 version:3 modified_at_second:1689777349
          volume id:11 size:8 replica_placement:1 version:3 modified_at_second:1689775114
          volume id:12 size:1675344 file_count:17 replica_placement:1 version:3 modified_at_second:1689777363
        Disk hdd total size:580287768 file_count:162
      DataNode 192.168.40.44:8080 total size:580287768 file_count:162
      DataNode 192.168.40.45:8080 hdd(volume:8/8 active:8 free:0 remote:0)
        Disk hdd(volume:8/8 active:8 free:0 remote:0)
          volume id:2 size:95268616 file_count:60 replica_placement:1 version:3 compact_revision:1 modified_at_second:1689777329
          volume id:4 size:103994728 file_count:60 replica_placement:1 version:3 compact_revision:1 modified_at_second:1689777306
          volume id:5 size:87643744 file_count:58 replica_placement:1 version:3 compact_revision:1 modified_at_second:1689777346
          volume id:7 size:136955864 file_count:2 replica_placement:1 version:3 modified_at_second:1689775203
          volume id:8 size:2047816 file_count:17 replica_placement:1 version:3 modified_at_second:1689777348
          volume id:9 size:8 replica_placement:1 version:3 modified_at_second:1689775114
          volume id:10 size:1400560 file_count:12 replica_placement:1 version:3 modified_at_second:1689777350
          volume id:12 size:1675344 file_count:17 replica_placement:1 version:3 modified_at_second:1689777363
        Disk hdd total size:428986680 file_count:226
      DataNode 192.168.40.45:8080 total size:428986680 file_count:226
      DataNode 192.168.40.62:8080 hdd(volume:5/8 active:5 free:3 remote:0)
        Disk hdd(volume:5/8 active:5 free:3 remote:0)
          volume id:1 size:106115208 file_count:66 replica_placement:1 version:3 compact_revision:1 modified_at_second:1689846063
          volume id:2 size:95268616 file_count:60 replica_placement:1 version:3 compact_revision:1 modified_at_second:1689777329
          volume id:4 size:103994728 file_count:60 replica_placement:1 version:3 compact_revision:1 modified_at_second:1689777305
          volume id:5 size:87643744 file_count:58 replica_placement:1 version:3 compact_revision:1 modified_at_second:1689777346
          volume id:6 size:618491832 file_count:501 delete_count:214 deleted_byte_count:44542857 replica_placement:1 version:3 modified_at_second:1689864456
        Disk hdd total size:1011514128 file_count:745 deleted_file:214 deleted_bytes:44542857
      DataNode 192.168.40.62:8080 total size:1011514128 file_count:745 deleted_file:214 deleted_bytes:44542857
    Rack rack-01 total size:2020788576 file_count:1133 deleted_file:214 deleted_bytes:44542857
  DataCenter datacenter-01 total size:2020788576 file_count:1133 deleted_file:214 deleted_bytes:44542857
total size:2020788576 file_count:1133 deleted_file:214 deleted_bytes:44542857

**Comment by htsnorman** [comment]
After a complete and proper restart all services, the replica was restored, and everything is functioning well. We conducted a rather drastic test during which I detached and reattached the volume disk offered to the process, and that's probably where the problem occurred, causing a loop in the fixed process. So, this part is sorted out, and it's very unusual for such an issue to happen.

However, the fuse-side mount continuously shows incorrect data regarding the used storage space. It didn't understand how it functions. I tried deleting and copying files, but at one point, when there were around 20 gigabytes of usage, it showed only 7 gigabytes in 'df' (disk free). Now, with approximately 2 gigabytes of usage, it shows 5.1 GB.

![CleanShot 2023-07-21 at 13 14 05](https://github.com/seaweedfs/seaweedfs/assets/9657774/4c0a18f3-e89b-4ce6-9194-60f364750a50)

thx @chrislusf
